pyweatherflowudp/mixins.py

```diff
@@ -0,0 +1,253 @@
+        return self._battery * UNIT_VOLTS
+
+    @property
+    def last_report(self) -> datetime | None:
+        """Return last report."""
+        return utc_timestamp_from_epoch(self._last_report)
+
+    @property
+    def report_interval(self) -> Quantity[int]:
+        """Return the report interval in minutes."""
+        return self._report_interval * UNIT_MINUTES
+
+
+class AirSensorMixin(BaseSensorMixin):
+    """Air sensor mixin."""
+
+    _air_temperature: float = 0
+    _last_lightning_strike_event: LightningStrikeEvent | None = None
+    _lightning_strike_average_distance: float = 0
+    _lightning_strike_count: int = 0
+    _relative_humidity: float = 0
+    _station_pressure: float = 0
+
+    @property
+    def air_temperature(self) -> Quantity[float]:
+        """Return the air temperature in degrees Celsius (°C)."""
+        return self._air_temperature * UNIT_DEGREES_CELSIUS
+
+    @property
+    def last_lightning_strike_event(self) -> LightningStrikeEvent | None:
+        """Return the last lightning strike event."""
+        return self._last_lightning_strike_event
+
+    @property
+    def lightning_strike_average_distance(
+        self,
+    ) -> Quantity[float]:
+        """Return the lightning strike average distance in kilometers (km)."""
+        return self._lightning_strike_average_distance * UNIT_KILOMETERS
+
+    @property
+    def lightning_strike_count(self) -> int:
+        """Return the lightning strike count."""
+        return self._lightning_strike_count
+
+    @property
+    def relative_humidity(self) -> Quantity[float]:
+        """Return the relative humidity percentage."""
+        return self._relative_humidity * UNIT_PERCENT
+
+    @property
+    def station_pressure(self) -> Quantity[float]:
+        """Return the station pressure in millibars (mbar)."""
+        return self._station_pressure * UNIT_MILLIBARS
+
+    # Derived metrics
+
+    @property
+    def air_density(self) -> Quantity[float]:
+        """Return the calculated air density in kilograms per cubic meter (kg/m³)."""
+        return mpcalc.density(self.station_pressure, self.air_temperature, 0)
+
+    @property
+    def delta_t(self) -> Quantity[float]:
+        """Return the calculated delta t in delta degrees Celsius (Δ°C)."""
+        return self.air_temperature - self.wet_bulb_temperature
+
+    @property
+    def dew_point_temperature(self) -> Quantity[float]:
+        """Return the calculated dew point temperature in degrees Celsius (°C)."""
+        return mpcalc.dewpoint_from_relative_humidity(
+            self.air_temperature, self.relative_humidity
+        )
+
+    @property
+    def heat_index(self) -> Quantity[float]:
+        """Return the calculated heat index in degrees Celsius (°C)."""
+        return mpcalc.heat_index(
+            self.air_temperature, self.relative_humidity, mask_undefined=False
+        )[0].to(UNIT_DEGREES_CELSIUS)
+
+    def sea_level_pressure(self, height: Quantity[float]) -> Quantity[float]:
```

Comment:
Missing a property tag?

Reply:
sea_level_pressure is the odd one out since it requires a height to properly calculate, and as such is actually a function. I should probably move that below the other properties and/or think of a different name for it to indicate it is not a property.

Reply:
Got it. Thanks for the clarification.

Reply:
Renamed to `calculate_sea_level_pressure`